File: MRDistribuidora/mrotica/exibicao/frmAviso.vb
Imports System.Windows.Forms
Public Class frmAviso
    Public Enum tipo_aviso
        tipo_ok = 1
        tipo_sim_nao = 2
    End Enum
    Public Enum respo
        SIM = 1
        NAO = 2
        OK = 3
        Cancel = 4
    End Enum
    Public tipo As New tipo_aviso
    Public resposta As respo = respo.OK
    Public alarme As Boolean = False
    Public Overloads Sub Show(ByVal msg As String)
        lblMSg.MaximumSize = Me.Size
        lblMSg.Text = msg
        Me.Show()
    End Sub
    Private Sub frmAviso_KeyDown(ByVal sender As Object, ByVal e As System.Windows.Forms.KeyEventArgs) Handles Me.KeyDown
        Select Case e.KeyCode
            Case Keys.Enter
                som_erro()
        End Select
    End Sub
    Private Sub frmAviso_Load(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles MyBase.Load
        lblMSg.MaximumSize = Me.Size
        Select Case tipo
            Case tipo_aviso.tipo_sim_nao
                btnNao.Visible = True
                btnSim.Visible = True
                btnEmerg.Focus()
            Case tipo_aviso.tipo_ok
                If alarme = True Then
                    som_erro()
                End If
                btnNao.Visible = False
                btnSim.Visible = False
                btnOk.Visible = True
                btnOk.Focus()
        End Select

    End Sub
    Private Sub som_erro()
        My.Computer.Audio.Play(Application.StartupPath & "\sounds\alertk2.wav")
    End Sub
    Private Sub btnSim_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnSim.Click
        resposta = respo.SIM
        Me.Close()
    End Sub
    Private Sub btnNao_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnNao.Click
        resposta = respo.NAO
        Me.Close()
    End Sub
    Private Sub btnOk_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnOk.Click
        resposta = respo.OK
        Me.Close()

    End Sub
    Private Sub btnEmerg_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles btnEmerg.Click
        som_erro()
    End Sub
End Class

File: MRSystem/MRUtil/Formulario/frmAviso.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using System.Windows.Forms;

namespace MRUtil.Formulario
{
    public partial class FrmAviso : Form
    {
        public Belemtech.TipoAviso tipo;
        public Belemtech.Respo resposta = Belemtech.Respo.OK;
        public Belemtech.ImagemAviso imgAviso;
        public FrmAviso()
        {
            InitializeComponent();
        }

        private void FrmAviso_Load(object sender, EventArgs e)
        {
            lblMensagem.MaximumSize = this.Size;
            switch (tipo)
            {
                case Belemtech.TipoAviso.SIMNAO:
                    BtnSim.Visible = true;
                    BtnSim.Location = new Point(226, 8);
                    BtnNao.Visible = true;
                    BtnNao.TabIndex = 0;
                    BtnOK.Visible = false;
                    break;
                case Belemtech.TipoAviso.OK:
                    BtnSim.Visible = false;
                    BtnNao.Visible = false;
                    BtnOK.Visible = true;
                    BtnOK.Location = new Point(302, 8);
                    BtnOK.Focus();
                    break;
            }

            try
            {
                if (imgAviso == Belemtech.ImagemAviso.Informacao)
                {
                    this.Icon = new System.Drawing.Icon(imageList1.Images.Keys[0]);
                }
                else if (imgAviso == Belemtech.ImagemAviso.Advertencia)
                {
                    this.Icon = new System.Drawing.Icon(imageList1.Images.Keys[1]);
                }
                else if (imgAviso == Belemtech.ImagemAviso.Error)
                {
                    this.Icon = new System.Drawing.Icon(imageList1.Images.Keys[2]);
                }
            }
            catch (System.IO.FileNotFoundException ex)
            {
                this.Icon = this.Icon;
            }

        }

        private void BtnSim_Click(object sender, EventArgs e)
        {
            resposta = Belemtech.Respo.SIM;
            this.Close();
        }

        private void BtnOK_Click(object sender, EventArgs e)
        {
            resposta = Belemtech.Respo.OK;
            this.Close();
        }

        private void BtnNao_Click(object sender, EventArgs e)
        {
            resposta = Belemtech.Respo.NAO;
            this.Close();
        }
    }
}
